### PR TITLE
IBM Power and Z documentation for Quay

### PR DESCRIPTION
**Pull request by alishaIBM**
Jira: https://issues.redhat.com/browse/PROJQUAY-6452

### CODE REVIEW

File: modules/config-ui-custom-ssl-certs.adoc [hunk 1]
```diff
@@ -2,6 +2,11 @@
 [id="config-ui-custom-ssl-certs"]
 = Custom SSL/TLS certificates UI
 
+[NOTE]
+====
+Config Editor UI has been removed and is not supported in IBM Power and IBM Z. Alternatively, Custom SSL/TLS certificates can be provided through config.yaml manually.
```
Comment (stevsmit) [inline modules/config-ui-custom-ssl-certs.adoc:7]:
```suggestion
The Config Editor UI has been removed and is not supported in IBM Power and IBM Z. Alternatively, Custom SSL/TLS certificates can be provided through your `config.yaml` manually.
```

File: modules/fips-overview.adoc [hunk 2]
```diff
@@ -17,7 +17,8 @@ Use the following procedure to enable FIPS compliance on your {productname} depl
 
 * If you are running a standalone deployment of {productname}, your {rhel} deployment is version 8 or later and FIPS-enabled. 
 * If you are using the {productname} Operator, {ocp} is version 4.10 or later.
-* Your {productname} version is 3.5.0 or later.  
+* Your {productname} version is 3.5.0 or later.
+* If you are using the {productname} Operator on IBM Power or IBM Z, {ocp} is version 4.14 or later and {productname} version is 3.10 or later is required.
```
Comment (stevsmit) [inline modules/fips-overview.adoc:21]:
```suggestion
* If you are using the {productname} Operator on IBM Power or IBM Z:
** {ocp} version 4.14 or later is required 
** {productname} version 3.10 or later is required
```

File: modules/georepl-intro.adoc [hunk 3]
```diff
@@ -2,6 +2,11 @@
 [id="georepl-intro"]
 = Geo-replication
 
+[NOTE]
+====
+Currently, this feature is not supported on IBM Power and IBM Z.
```
Comment (stevsmit) [inline modules/georepl-intro.adoc:7]: For consistency with the rest of the notes: 

```suggestion
Currently, the geo-replication feature is not supported on IBM Power and IBM Z.
```

File: modules/operator-config-ui-access.adoc [hunk 4]
```diff
@@ -2,6 +2,11 @@
 [id="operator-config-ui-access"]
 = Accessing the config editor
 
+[NOTE]
+====
+Config Editor UI has been removed and is not supported in IBM Power and IBM Z.
```
Comment (stevsmit) [inline modules/operator-config-ui-access.adoc:7]: The rest say "on"

```suggestion
The Config Editor UI has been removed and is not supported on IBM Power and IBM Z.
```

File: modules/operator-customize-images.adoc [hunk 5]
```diff
@@ -2,6 +2,11 @@
 [id="operator-customize-images"]
 = Customizing Default Operator Images
 
+[NOTE]
+====
+Currently, customizing default operator images is not supported on IBM Power and IBM Z.
```
Comment (stevsmit) [inline modules/operator-customize-images.adoc:7]:
```suggestion
Currently, customizing default Operator images is not supported on IBM Power and IBM Z.
```

File: modules/rn_3_10_0.adoc [hunk 6]
```diff
@@ -15,6 +15,11 @@ For more information, see the link:https://access.redhat.com/support/policy/upda
 [id="new-features-and-enhancements-310"]
 == {productname} new features and enhancements
 
+[id="ibm-power-z-linuxone-support"]
+===== IBM Power, IBM Z, IBM® LinuxONE support:
```
Comment (stevsmit) [inline modules/rn_3_10_0.adoc:19]: Remove one =

```suggestion
==== IBM Power, IBM Z, IBM® LinuxONE support:
```

File: modules/proc_setting-up-quay-for-qbo.adoc [hunk 7]
```diff
@@ -30,7 +30,7 @@ In this procedure, you will create a dedicated {productname} organization, and f
 
 . Review the assigned permissions.
 
-. Select *Authorize Application* and then confirm confirm the authorization by selecting *Authorize Application*.
+. Select *Authorize Application* and then confirm the authorization by selecting *Authorize Application*.
```
Comment (yussufsh) [inline modules/proc_setting-up-quay-for-qbo.adoc:33]: Can this go in another PR?

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: modules/builders-virtual-environment.adoc [hunk 8]
```diff
@@ -104,7 +104,7 @@ eyJhbGciOiJSUzI1NiIsImtpZCI6IldfQUJkaDVmb3ltTHZ0dGZMYjhIWnYxZTQzN2dJVEJxcDJscldS
 +
 [source,terminal]
 ----
-$ oc get route -n quay-enterprise
+$ oc get route -n quay-enterprise —duration
```
Comment (yussufsh) [inline modules/builders-virtual-environment.adoc:107]: There is no such option for `oc get route`. Please remove.

File: modules/config-ui-custom-ssl-certs.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (dongboyan77) [reply modules/config-ui-custom-ssl-certs.adoc:7]: Config Editor will be removed in quay operator for all arch. But customer still can setup a config editor standalone via podman/docker run container. So this field needs to describe more exact

**Review by juliemathew — COMMENTED** [review]
The following features should be mentioned as 'unsupported'  in the RN for 3.10 under  "known issues and limitations" section, for P and Z.
1.  Geo-Replication
2.  IPv6 Single stack/ Dual Stack 
3. Mirror registry
4. Quay config editor -  Mirror, MAG, Kinesis, Keystone, GitHub Enterprise, OIDC 

@alishaIBM 
cc: @yussufsh @modassarrana89

File: modules/operator-config-ui-access.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (modassarrana89) [reply modules/operator-config-ui-access.adoc:7]: This is true for all arch , but for x86 , they can mention to use old version which is not the case for P & Z

File: modules/builders-virtual-environment.adoc [hunk 9]
```diff
@@ -91,7 +91,7 @@ oc sa get-token -n virtual-builders quay-builder
 +
 [source,terminal]
 ----
-$ oc create token quay-builder -n virtual-builders
+$ oc create token quay-builder -n virtual-builders —duration
```
[the remark this answers is not in the document]
Comment (ksdeekshith) [reply modules/builders-virtual-environment.adoc:94]: It's a generic change, whenever the token generated at that step, it's not sticking to that for longer, so I had to recreate multiple times, to resolve that I suggested to add that `--duration` not `-duration` flag along with time in minutes as units (which is supported)

File: modules/builders-virtual-environment.adoc (hunk 8 above)
Comment (modassarrana89) [reply modules/builders-virtual-environment.adoc:107]: Remove this option , as Yussuf has suggested

**Comment by modassarrana89** [comment]
@alishaIBM v2 user interface is not supported for Z . @yussufsh @juliemathew Please confirm from Power side .

File: modules/rn_3_10_0.adoc (hunk 6 above)
Comment (alishaIBM) [reply modules/rn_3_10_0.adoc]: Thank you @stevsmit 
I have incorporated the suggestions into the PR.

File: modules/builders-virtual-environment.adoc (hunk 8 above)
Comment (alishaIBM) [reply modules/builders-virtual-environment.adoc]: Done. Thanks.

File: modules/proc_installing-qbo-on-ocp.adoc [hunk 10]
```diff
@@ -4,7 +4,7 @@
 
 In this procedure, you will install the {qbo} on {ocp}.
 
-.Prerequiites
+.Prerequisites
```
[the remark this answers is not in the document]
Comment (alishaIBM) [reply modules/proc_installing-qbo-on-ocp.adoc:7]: okay.

File: modules/operator-config-ui-access.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (alishaIBM) [reply modules/operator-config-ui-access.adoc]: @dmesser could you please check above ?

File: modules/config-ui-custom-ssl-certs.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (alishaIBM) [reply modules/config-ui-custom-ssl-certs.adoc]: I will need more info (content) here. @dongboyan77 @yussufsh @dmesser

File: modules/builders-virtual-environment.adoc (hunk 9 above)
[the remark this answers is not in the document]
Comment (alishaIBM) [reply modules/builders-virtual-environment.adoc:94]: @ksdeekshith could you please provide the exact command here ? Thanks.

[the remark this answers is not in the document]
Comment (SNiemann15) [reply modules/builders-virtual-environment.adoc:94]: @alishaIBM

Comment (SNiemann15) [inline modules/builders-virtual-environment.adoc:94]:
```suggestion
$ oc create token quay-builder -n virtual-builders --duration
```

File: modules/operator-config-ui-access.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (stevsmit) [reply modules/operator-config-ui-access.adoc]: I just updated this. It's not as intricate as it sounds. Simply put: the config editor on the Red Hat Quay Operator was removed for security reasons. The pod no longer spins up, and the endpoint no longer generates on the UI. My PR for this is here: https://github.com/quay/quay-docs/pull/842/files

File: modules/builders-virtual-environment.adoc (hunk 9 above)
Comment (ksdeekshith) [reply modules/builders-virtual-environment.adoc:94]: @SNiemann15  can we add `--duration=20160m` as default for 2weeks. And a note like as can be changed as per requirement.

Comment (dmage) [reply modules/builders-virtual-environment.adoc:94]: `--duration` needs an argument, something like `--duration 10m`

Comment (dmage) [reply modules/builders-virtual-environment.adoc:94]: @SNiemann15 it can be with or without `=`,  `--duration 10m` is an example from `oc create token --help`.

`<time_in_minutes>` isn't ideal name, if you run it with `--duration 10`, it'll fail with the message:

```
error: invalid argument "10" for "--duration" flag: time: missing unit in duration "10"
```

I.e. you have to specify units, and it can be seconds, minutes, or hours. For example:

* `--duration=336h`
* `--duration=20160m`
* `--duration=1209600s`
* `--duration=335h60m`

All these options give the same result.

When the option is not specified, the tokens used not to expire. This might have changed in recent OCP versions though.

File: modules/config-ui-custom-ssl-certs.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (alishaIBM) [reply modules/config-ui-custom-ssl-certs.adoc]: Okay. Removed the note. The file is intact.

File: modules/builders-virtual-environment.adoc [hunk 11]
```diff
@@ -91,8 +91,9 @@ oc sa get-token -n virtual-builders quay-builder
 +
 [source,terminal]
 ----
-$ oc create token quay-builder -n virtual-builders
+$ oc create token quay-builder -n virtual-builders --duration=<time_in_minutes> <1>
 ----
+<1> Optional. Specify the duration, to request a token with a custom expiration. The value and the unit are required. For example, specify `--duration 20160m` to retain the token for two weeks.
```
Comment (SNiemann15) [inline modules/builders-virtual-environment.adoc:96]: @alishaIBM I discussed with @dmage to change as follows: 

```suggestion
$ oc create token quay-builder -n virtual-builders
----
+
[NOTE]
====
When the token expires you will need to request a new token. Optionally, you can also  add a custom expiration. For example, specify `--duration 20160m` to retain the token for two weeks.
====
```

Comment (alishaIBM) [reply modules/builders-virtual-environment.adoc]: Thank you @SNiemann15. I have updated the PR.

**Comment by yussufsh** [comment]
> @alishaIBM RedHat Quay V2 user interface & HA Deployment is not supported for Z . Add this in the RN 3.10 under "Known issues and limitations" @yussufsh @juliemathew Please confirm from Power side .

For IBM Power:
1. `RedHat Quay V2 User Interface` is not supported (this can go in common P/Z statement)
2. `Deploy Red Hat Quay - High Availability` is supported for P but the following is not:
   - Back up and restore
   - Migrating a standalone to operator

**Comment by alishaIBM** [comment]
@yussufsh could you approve for IBM Power

File: modules/rn_3_10_0.adoc [hunk 12]
```diff
@@ -15,6 +15,11 @@ For more information, see the link:https://access.redhat.com/support/policy/upda
 [id="new-features-and-enhancements-310"]
 == {productname} new features and enhancements
 
+[id="ibm-power-z-linuxone-support"]
+==== IBM Power, IBM Z, IBM® LinuxONE support:
+
+With this release, IBM Power (ppc64le), IBM Z (390x), and IBM® LinuxONE (s390x) architectures are supported.
```
Comment (modassarrana89) [inline modules/rn_3_10_0.adoc:21]: IBMZ (390x) is wrong. it will be IBM Z(s390x)

File: modules/rn_3_10_0.adoc [hunk 13]
```diff
@@ -99,6 +104,16 @@ The following sections note known issues and limitations for {productname} {prod
 [id="limitations-310"]
 === {productname} 3.10 limitations
 
+* In this release, the following features are not supported in IBM Power (ppc64le) and IBM Z (390x):
```
Comment (modassarrana89) [inline modules/rn_3_10_0.adoc:107]: IBMZ (390x) is wrong. it will be IBM Z(s390x)

File: modules/rn_3_10_0.adoc (hunk 12 above)
Comment (alishaIBM) [reply modules/rn_3_10_0.adoc]: Ack
